File: Java/프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int gcd(int a, int b) { // 유클리드 호제법
        if(b == 0) return a;
        return gcd(b, a % b); // 재귀함수 (a % b)를 계속 하다가 (a % b)가 0이 되면 b 리턴)
    }
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int[] answer = new int[2]; // 분모, 분자만 나오니까 배열 길이는 2
        int a = (numer1 * denom2) + (numer2 * denom1); // 분자
        int b = denom1 * denom2; // 분모
        
        int g = gcd(a,b); // 최대 공약수
        
        answer[0] = a / g;
        answer[1] = b / g;
        return answer;
    }
}

File: JavaScript/프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.js
function num (a,b) {
    if(b === 0) return a;
    return num(b, a%b);
}
function solution(numer1, denom1, numer2, denom2) {
    const num1 = ((numer1 * denom2) + (numer2 * denom1));
    const num2 = denom1 * denom2;
    
    let number = num(num1, num2);
    return [num1/number, num2/number];
}